src/lib/agent/MessageReceiver.ts

```diff
@@ -21,20 +23,26 @@ class MessageReceiver {
     if (!inboundPackedMessage['@type']) {
       inboundMessage = await this.envelopeService.unpackMessage(inboundPackedMessage);
 
-      if (!inboundMessage.message['@type']) {
+      // TODO: why is it different than before?
+      // TODO: we should determine whether we are the target of the forward message
+      // if we are we should unpack the msg from the forward message and handle it
+      // if we are not, and we should forward, we do not unpack and pass the forward message to the forward handler
+      if (inboundMessage.message['@type'] === MessageType.ForwardMessage && this.config.agencyUrl !== undefined) {
```

Comment:
@jakubkoci Could you give some clarification on this piece of code?

In the previous implementation the message got unpacked. If the message didn't contain an `@type` it would be unpacked again. However thinking about it this doesn't seem like the correct approach, and i'm not sure why it worked.

To my understanding we should
- unpack the received message
    - if the `@type` of the unpacked message is not the type of a forward message pass the message as is to the dispatcher and handle like any message
    - if the `@type` of the unpacked message is of type forward
         - and the `to` field is a key belonging to us, we should unpack the `msg` field of the forward message and pass it to the dispatcher and handle like any message
         - if the `to` field is not a key belonging to us, we should not unpack the `msg` field and pass the forward message to the dispatcher, which will then be handled by the forward handler.

The check whether the `to` field is a key belonging to us could be done in two ways i think. The first is just trying to unpack, if we can unpack the message we certainly are the owner of the key in the `to` field. The other is first checking whether the key belongs to us and only unpacking if so. First will be easier.

So this implementation is not correct and will need to be updated before merging. However the exact process of handling forward message for both edge and mediator is still somewhat confusing to me, so I want to have mutual understanding of the process first.

What do you think?